components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/AbstractApplicationMgtListener.java

```diff
@@ -420,7 +420,7 @@ public boolean doPreUpdateApplicationTemplate(ServiceProvider serviceProvider, S
     public boolean doPostDeleteApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
             throws IdentityApplicationManagementException {
 
-        return true;
+        return doPostDeleteApplication(serviceProvider.getApplicationName(), tenantDomain, userName);
```

Comment:
This change is required since there might be legacy listeners who have only implemented 
```
doPostDeleteApplication(serviceProvider.getApplicationName(), tenantDomain, userName)
```

Now the service layer calls,
```
public boolean doPostDeleteApplication(ServiceProvider serviceProvider, String tenantDomain, String userName)
            throws IdentityApplicationManagementException
```

If we do not call the legacy method then we might miss executing post delete actions of legacy listeners

Reply:
Have we deprecated the legacy one? Someone may implement both which will result in repeating the same work

Reply:
Yes, we have deprecated it in https://github.com/wso2/carbon-identity-framework/blob/7c151822bfc72d3d9b3c2bd3056cd98e0b73dd0e/components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtListener.java#L123

Reply:
Seems it's not done at AbstractApplicationMgtListener. That's why I added the comment initially. Someone is more likely to use Abstract class instead of interface

Reply:
Agreed. Added depracted annotation and javadoc to both interface and abstract layer